src/components/CreateCriteriaForm/CreateCreteriaForm.tsx

```diff
@@ -0,0 +1,272 @@
+
+import { AddCriteriaScheme, criteriaSchema } from '../../schema/criteria';
+import { Table, TableRow, TitleItem, ContentItem, Title, TextItem, TitleContainer } from '../Table';
+import { StateDot } from '../StateDot';
+import { errorsProvider } from '../../utils/forms';
+
+import { tr } from './CreateCriteriaForm.i18n';
+
+const maxPossibleWeigth = 100;
+
+const StyledPlainButton = styled(Button)`
+    background-color: unset;
+    border: none;
+    padding: 0;
+
+    color: ${gray8};
+
+    border-radius: 0;
+
+    &:hover:not([disabled]),
+    &:focus:not([disabled]) {
+        background-color: unset;
+    }
+
+    &:active:not([disabled]) {
+        transform: none;
+    }
+`;
+
+const StyledTableResults = styled(Table)`
+    grid-template-columns: 30px 210px 40px minmax(max-content, 120px);
+    width: fit-content;
+`;
+
+const StyledFormInput = styled(FormInput)`
+    font-size: 14px;
+    font-weight: normal;
+    padding: 5px 10px;
+
+    border: 1px solid ${gray7};
+    box-sizing: border-box;
+
+    & ~ div {
+        top: 50%;
+    }
+`;
+
+const StyledTableRow = styled.div`
+    display: grid;
+    padding-left: 14px;
+    grid-template-columns:
+        minmax(calc(300px + 10px /* gap of sibling table */), 30%)
+        repeat(2, max-content);
+`;
+
+const StyledSubmitButton = styled(Button)`
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+`;
+
+const GoalSuggestItem = (props: any): React.ReactElement => {
+    const handleClick = (event: any) => {
+        event.preventDefault();
+        props.onClick();
+    };
+
+    return (
+        <TableRow href="#" onClick={handleClick} focused={props.focused}>
+            <ContentItem>
+                <GoalIcon size="s" />
+            </ContentItem>
+            <TitleItem>
+                <TitleContainer>
+                    <Title>{props.title}</Title>
+                </TitleContainer>
+            </TitleItem>
+            <ContentItem>
+                {nullable(props.state, (s) => (
+                    <StateDot size="m" title={s?.title} hue={s?.hue} />
+                ))}
+            </ContentItem>
+            <ContentItem>
+                <TextItem>{props.projectId}</TextItem>
+            </ContentItem>
+        </TableRow>
+    );
+};
+
+interface WeightFieldProps {
+    registerProps: UseFormRegisterReturn<'weight'>;
+    errorsResolver: (field: 'weight') => { message?: string } | undefined;
+    maxValue: number;
+    setError: UseFormSetError<AddCriteriaScheme>;
+}
+
+const WeightField: React.FC<WeightFieldProps> = ({ registerProps, errorsResolver, maxValue, setError }) => {
+    const handleChange = useCallback<React.ChangeEventHandler<HTMLInputElement>>(
+        (event) => {
+            const { value } = event.target;
+
+            const parsedValue = parseInt(value, 10);
+            let message: string | undefined;
+
+            if (!value) {
```

Comment:
All of this validations is possible to do with zod schema

Reply:
Also, we need to check existing criteria with same title. Title is uniq but in scope of goal.